components/user_profile/user_profile.tsx

```diff
@@ -54,6 +61,29 @@ export default class UserProfile extends PureComponent<UserProfileProps> {
         this.overlay = ref;
     }
 
+    generateColor = (username: string, background: string): string => {
+        const cacheKey = `${username}-${background}`;
+        const cachedColor = LocalStorageStore.getItem(cacheKey);
+        if (cachedColor !== null) {
+            return cachedColor;
+        }
+
+        let userColor = background;
+        let userAndSalt = username;
+        const checker = new ColorContrastChecker();
+        const colorHash = new ColorHash();
+
+        let tries = 10;
```

Comment:
Personally I'd be in favour of less tries here (2-3 would be my feeling) but I'm not sure to what degree that will affect performance.

I'm happy with the change to this though, so only 1/5 on the actual number.

Reply:
3 sounds good.